src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java

```diff
@@ -113,63 +117,134 @@
 
     /**
      * Creates a FeatureDataSource backed by the provided file that uses the given codec to decode records
```

Comment:
"the given codec" is no longer applicable, since you've internalized codec discovery